src/crimson/osd/pg.cc

```diff
@@ -929,9 +929,9 @@ PG::do_osd_ops_execute(
   FailureFunc&& failure_func)
 {
   assert(ox);
-  auto rollbacker = ox->create_rollbacker([this] (auto& obc) {
-    return obc_loader.reload_obc(obc).handle_error_interruptible(
-      load_obc_ertr::assert_all{"can't live with object state messed up"});
+  auto rollbacker = ox->create_rollbacker([object_context=*obc] (auto& obc) {
+    obc = object_context;
```

Comment:
The new (copied) obc won't be in `obc_registry`, right? This can be a bit tricky for future cache usages. Perhaps we should handle this case as well? Or can we assume the the copied obc should not be cached and may be only used on the failure path.

Reply:
> Or can we assume the the copied obc should not be cached and may be only used on the failure path.

Yeah, I think we can have that assumption.